modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradePortletDisplayCheck.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradePortletDisplayCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String afterFormat(
+		String fileName, String absolutePath, String content,
+		String newContent) {
+
+		if (fileName.endsWith(".jsp")) {
+			newContent = addNewImportsJSP(
+				newContent,
+				"com.liferay.portal.configuration.module.configuration." +
+					"ConfigurationProviderUtil");
+		}
+		else {
+			newContent = addNewImports(newContent);
+		}
+
+		return newContent;
+	}
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		String newContent = content;
+
+		if (fileName.endsWith(".jsp")) {
+			return _formatJSP(content);
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
```

Comment:
@TamyrisTorres, can't we reuse part of the code to format the different files, Java and JSP?

It seems that part of them can be reused, actually, the JSP formatting seems to be a subset of the Java formatting, is this correct?

Reply:
I need to use javaTerm to limit the scope and get the variable "themeDisplay", so because of this, I use "javaMethodContent" for matcher when the file is ".java". When is JSP the matcher uses the content. I can't use the javaTerm in the ".jsp" file.